Firestore/core/src/firebase/firestore/local/leveldb_remote_document_cache.cc

```diff
@@ -73,9 +73,10 @@ class AsyncResults {
     values_.push_back(value);
   }
 
-  const std::vector<T>& Result() {
+  std::vector<T> Result() {
     std::lock_guard<std::mutex> lock(mutex_);
-    return values_;
+    std::vector<T> result = std::move(values_);
```

Comment:
A couple of final nitpicks:
1. I had an intermediate variable in my code example only so that the member variable could be `clear`ed before exiting the function. If you don't think it's worthwhile, then it could be just `return std::move(values_)`.
2. Can you please add a comment to this function explaining either that it clears the vector or that the object is not supposed to be used again after `Result` is called on it, depending on whichever aligns more with your intention?